src/adaptors/stargate-finance/index.js

```diff
@@ -0,0 +1,219 @@
+
+    return {
+        lpToken,
+        lpTokenSymbol,
+        underlyingLpToken,
+        allocPoint,
+        totalAllocPoint,
+        rewardPerBlock,
+        reserve,
+    };
+}
+
+
+const getPrices = async (chain, addresses) => {
+
+    const prices = (
+        await superagent.post('https://coins.llama.fi/prices').send({
+            coins: addresses.map((address) => `${chain}:${address}`),
+        })
+    ).body.coins;
+
+    const pricesObj = Object.entries(prices).reduce(
+        (acc, [address, price]) => ({
+            ...acc,
+            [address.split(':')[1].toLowerCase()]: price.price,
+        }),
+        {}
+    );
+
+    return pricesObj;
+};
+
+
+const tvl = async (chain, symbol, underlyingLpToken, reserve) => {
+    // total number of coins in pool * coin price
+    let token = underlyingLpToken;
+    if (symbol === 'S*SGETH') {
+        token = CONFIG[chain].ETHER_TOKEN;
+    }
+
+    const price = (await getPrices(chain, [token.toLowerCase()]))[token.toLowerCase()];
+    const reserveUSD = reserve * price;
+
+    return reserveUSD;
+}
+
+
+function calcApy(chain, allocPoint, totalAllocPoint, reward, rewardPrice, reserve) {
+    // pool rewards per year in usd
+    // blocks per year * reward * wieght * price
+
+    // BLOCK_TIME is number of seconds per 1 block settled
+    // get avg block time directly from provider?
+    let BLOCK_TIME = 13.5;
+    if (chain == 'fantom') {
+        BLOCK_TIME = 1;
+    }
+    if (chain == 'optimism') {
+        BLOCK_TIME = 1;
+    }
+    if (chain == 'polygon') {
+        BLOCK_TIME = 2.11;
+    }
+    if (chain == 'avax') {
+        BLOCK_TIME = 2.03;
+    }
+    if (chain == 'bsc') {
+        BLOCK_TIME = 3;
+    }
+
+    const SECONDS_PER_YEAR = 60 * 60 * 24 * 365;
+    const BLOCKS_PER_YEAR = SECONDS_PER_YEAR / BLOCK_TIME;
+
+    const weight = allocPoint / totalAllocPoint;
+    const rewardPerBlock = reward * weight;
+
+    const rewardPerYear = rewardPerBlock / 1e18 * BLOCKS_PER_YEAR;
+
+    const rewardUSD = rewardPerYear * rewardPrice;
+    const apr = rewardUSD / reserve * 100;
+
+    return apr;
+
+}
+
+
+const getApy = async (chain, lpStaking) => {
+
+    let poolsApy = [];
+    const poolLength = parseInt((await sdk.api.abi.call({ abi: abi.poolLength, target: lpStaking, chain })).output);
+
+    let rewardToken = CONFIG[chain].STG_TOKEN;
+    if (chain == 'optimism') {
+        rewardToken = CONFIG[chain].OP_TOKEN;
+    }
+    const rewardPrice = (await getPrices(chain, [rewardToken.toLowerCase()]))[rewardToken.toLowerCase()];
+
+    for (i = 0; i < poolLength; i++) {
+        const pool = await pools(i, chain, lpStaking);
+        const reserveUSD = await tvl(chain, pool.lpTokenSymbol, pool.underlyingLpToken, pool.reserve);
+        const apy = calcApy(chain, pool.allocPoint, pool.totalAllocPoint, pool.rewardPerBlock, rewardPrice, reserveUSD);
+
+        poolsApy.push({
+            pool: pool.lpToken,
```

Comment:
the reason the test fails is because you are using the same id twice for 2 different objects.
all u need to do is: `${pool.lpToken}-${chain}`.toLowerCase() and it will be fine :)

Reply:
tysm! if im understanding your fix correctly this would have the pool property of the object look like dis
pool: '0x29e38769f23701a2e4a8ef0492e19da4604be62c-avax'
your fix passes the failed test
if this is the intended output ill commit the change : )

Reply:
ah now that u mentioned it, please change the chain names to the name field in here: `https://api.llama.fi/chains`
eg 
```
  {
    "gecko_id": "binancecoin",
    "tvl": 6469169987.419418,
    "tokenSymbol": "BNB",
    "cmcId": "1839",
    "name": "Binance",
    "chainId": 56
  },
```
instead of bsc pls use Binance as your `chain` value

Reply:
if i add `${pool.lpToken}-${chain}` or change the chain names thru my config const my code breaks in a major way...
i was thinking of sanitizing the objects in poolsApy by running it into a separate function
this seemed like the easiest solution but i know there has to be a cleaner way...

function sanitizeExport(poolsApy)

    const swapName1 = poolsApy.map(obj => obj.chain === "bsc" ? { ...obj, chain: 'Binance' } : obj);
    const swapPool1 = swapName1.map(obj => obj.chain === 'Binance' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName2 = swapPool1.map(obj => obj.chain === "avax" ? { ...obj, chain: 'Avalanche' } : obj);
    const swapPool2 = swapName2.map(obj => obj.chain === 'Avalanche' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName3 = swapPool2.map(obj => obj.chain === "ethereum" ? { ...obj, chain: 'Ethereum' } : obj);
    const swapPool3 = swapName3.map(obj => obj.chain === 'Ethereum' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName4 = swapPool3.map(obj => obj.chain === "polygon" ? { ...obj, chain: 'Polygon' } : obj);
    const swapPool4 = swapName4.map(obj => obj.chain === 'Polygon' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName5 = swapPool4.map(obj => obj.chain === "arbitrum" ? { ...obj, chain: 'Arbitrum' } : obj);
    const swapPool5 = swapName5.map(obj => obj.chain === 'Arbitrum' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName6 = swapPool5.map(obj => obj.chain === "optimism" ? { ...obj, chain: 'Optimism' } : obj);
    const swapPool6 = swapName6.map(obj => obj.chain === 'Optimism' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    const swapName7 = swapPool6.map(obj => obj.chain === "fantom" ? { ...obj, chain: 'Fantom' } : obj);
    const swapPool7 = swapName7.map(obj => obj.chain === 'Fantom' ? { ...obj, pool: `${obj.pool}-${obj.chain}` } : obj);

    return swapPool7;

Reply:
hmm, I'd keep it much simpler by just creating a new object in global scope (like `CONFIG`) which maps from your config keys (your chain names) to our chain names? that should solve it

Reply:
alrighty ty ill think about how i can do dat
the problemo is 
``superagent.post('https://coins.llama.fi/prices').send({
            coins: addresses.map((address) => `${chain}:${address}`),
   })``
requires chain = 'bsc' using 'Binance' will return undefined
similarly `sdk.api.abi.call({` will break if using 'Binance' and not 'bsc'
(if im not mistaken)
so i really really appreciate you helping me to think thru it

Reply:
im not entirely sure how i would map the keys into a new obj so i jus added a new entry in `CONFIG`
example:
```
const CONFIG = {
    bsc: {
        LP_STAKING: 0x3052A0F6ab15b4AE1df39962d5DdEFacA86DaB47,
        STG_TOKEN: 0xB0D502E938ed5f4df2E681fE6E419ff29631d62b,
        LLAMA_NAME: Binance,
    },
}
```

now i can access in poolsApy like dis (ur original suggestion)
```
poolsApy.push({
     pool: ${pool.lpToken}-${CONFIG[chain].LLAMA_NAME}.toLowerCase(),
     chain: CONFIG[chain].LLAMA_NAME
});
```

Reply:
yes, this is what i meant, niice. i think u have addresses everything

Reply:
🐋 ty for talking through it wit me : )  smh 🤦  y didnt i think of that sooner lol